File: 82_remove_duplicates_from_sorted_list_ii/remove_duplicates_from_sorted_list_ii.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */


struct ListNode* deleteDuplicates(struct ListNode* head){
    int ans[300];
    int size = 0;
    struct ListNode* node;
    if (head == NULL) {
        return head;
    } else {
        ans[size++] = head->val;
    }
    node = head->next;
    while (node) {
        if (size > 0 && node->val == ans[size - 1]) {
            while (node->next && node->next->val == node->val) {
                node = node->next;
            }
            size--;
        } else {
            ans[size++] = node->val;
        }
        node = node->next;
    }
    struct ListNode* dummy = malloc(sizeof(struct ListNode));
    dummy->val = 0;
    dummy->next = NULL;
    node = dummy;
    for (int i =0; i<size;i++) {
        struct ListNode* obj = malloc(sizeof(struct ListNode));
        obj->val = ans[i];
        obj->next = NULL;
        node->next = obj;
        node = obj;
    }
    return dummy->next;
}

File: 82_remove_duplicates_from_sorted_list_ii/remove_duplicates_from_sorted_list_ii.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


def list_to_node(nums: list) -> ListNode:
    """从列表构建链表"""
    dummy = ListNode()
    node = dummy
    for n in nums:
        node.next = ListNode(n)
        node = node.next
    return dummy.next


def node_to_list(head: ListNode) -> list:
    """从链表转列表"""
    result = []
    while head:
        result.append(head.val)
        head = head.next
    return result


class Solution:
    def deleteDuplicates(self, head: ListNode) -> ListNode:
        table = dict()
        node = head
        while node:
            table[node.val] = table.get(node.val, 0) + 1
            node = node.next
        dummy = cur = ListNode()
        for k, v in table.items():
            if v == 1:
                cur.next = ListNode(k)
                cur = cur.next
        return dummy.next
